### PR TITLE
Added Zolertia Zonik sound sensor test and drivers

### PR DESCRIPTION
**Pull request by amejias**
Drivers for the Zonik sonometer sensor. Added drivers and test application.

### CODE REVIEW

File: platform/zoul/dev/zonik.c [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      clock_delay_usec(ZONIK_FINAL_WAIT_DELAY);
+      i2c_master_enable();
+      if(i2c_burst_receive(ZONIK_ADDR, &zonik_buffer[0], ZONIK_FRAME_SIZE) != 
+                           I2C_MASTER_ERR_NONE) {
+        zonik_status = ZONIK_ERROR;
+        PRINTF("Zonik: Error in I2C Burst Mode Receive");
+      }
+      #if DEBUG
+      PRINTF("\nZonik: ");
+      for(i=0;i<ZONIK_FRAME_SIZE;i++) {
```
Comment (alignan) [inline platform/zoul/dev/zonik.c]: Code style, add a space between the elements inside the `for`

File: platform/zoul/dev/zonik.c [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        PRINTF(" 0x%02x ", zonik_buffer[i]);	  
+      }
+      PRINTF("\n");
+      #endif
+      if (zonik_status != ZONIK_ERROR) zonik_status = ZONIK_ACTIVE;
```
Comment (alignan) [inline platform/zoul/dev/zonik.c]: Code style, avoid this one-liner and replace as:

````
if(zonik_status != ZONIK_ERROR) {
  zonik_status = ZONIK_ACTIVE;
}
````

Comment (amejias) [inline platform/zoul/dev/zonik.c]: Done!

File: platform/zoul/dev/zonik.c (hunk 1 above)
Comment (amejias) [inline platform/zoul/dev/zonik.c]: Done!

**Comment by alignan** [comment]
:+1:

Let's wait for Travis

**Pull request merged**